controllers/renovate_util.go

```diff
@@ -268,27 +309,130 @@ func (u ComponentDependenciesUpdater) GetUpdateTargetsGithubApp(ctx context.Cont
+		if err != nil {
+			log.Error(err, "can't parse automerge option in configmap", "configMapName", customRenovateConfigMapName, "automergeValue", automergeOption)
+		} else {
+			customRenovateOptions.Automerge = automergeValue
+		}
+	}
+
+	platformAutomergeOption, platformAutomergeExists := customRenovateConfigMap.Data[RenovateConfigMapPlatformAutomergeKey]
+	if platformAutomergeExists {
+		platformAutomergeValue, err := strconv.ParseBool(platformAutomergeOption)
+		if err != nil {
+			log.Error(err, "can't parse platformAutomerge option in configmap", "configMapName", customRenovateConfigMapName, "platformAutomergeValue", platformAutomergeOption)
+		} else {
+			customRenovateOptions.PlatformAutomerge = platformAutomergeValue
+		}
+	}
+
+	ignoreTestsOption, ignoreTestsExists := customRenovateConfigMap.Data[RenovateConfigMapIgnoreTestsKey]
+	if ignoreTestsExists {
+		ignoreTestsValue, err := strconv.ParseBool(ignoreTestsOption)
+		if err != nil {
+			log.Error(err, "can't parse ignoreTests option in configmap", "configMapName", customRenovateConfigMapName, "ignoreTestsValue", ignoreTestsOption)
+		} else {
+			customRenovateOptions.IgnoreTests = ignoreTestsValue
+		}
+	}
+
+	automergeTypeOption, automergeTypeExists := customRenovateConfigMap.Data[RenovateConfigMapAutomergeTypeKey]
+	if automergeTypeExists {
+		customRenovateOptions.AutomergeType = automergeTypeOption
+	}
+
+	commitMessagePrefixOption, commitMessagePrefixExists := customRenovateConfigMap.Data[RenovateConfigMapCommitMessagePrefixKey]
+	if commitMessagePrefixExists {
+		customRenovateOptions.CommitMessagePrefix = commitMessagePrefixOption
+	}
+
+	commitMessageSuffixOption, commitMessageSuffixExists := customRenovateConfigMap.Data[RenovateConfigMapCommitMessageSuffixKey]
+	if commitMessageSuffixExists {
+		customRenovateOptions.CommitMessageSuffix = commitMessageSuffixOption
+	}
+
+	fileMatchOption, fileMatchExists := customRenovateConfigMap.Data[RenovateConfigMapFileMatchKey]
+	if fileMatchExists {
+		fileMatchParts := strings.Split(fileMatchOption, ",")
+		for i := range fileMatchParts {
+			fileMatchParts[i] = strings.TrimSpace(fileMatchParts[i])
+		}
+		customRenovateOptions.FileMatch = fileMatchParts
+	}
+
+	return &customRenovateOptions, nil
+}
+
+// readComponentCustomRenovateConfigMapAnnotation returns name of ConfigMap with custom renovate settings for nudge.
+// The ConfigMap name is taken from CustomRenovateConfigMapAnnotation annotation on the Component.
+func readComponentCustomRenovateConfigMapAnnotation(component *v1alpha1.Component) string {
+	if component.Annotations == nil {
+		return ""
+	}
+
+	customRenovateConfigMapAnnotation, customRenovateConfigMapAnnotationExists := component.Annotations[CustomRenovateConfigMapAnnotation]
+	if customRenovateConfigMapAnnotationExists && customRenovateConfigMapAnnotation != "" {
+		return customRenovateConfigMapAnnotation
+	}
+	return ""
+}
+
+// generateRenovateConfigForNudge returns renovate config for target
```

Comment:
please explain in the comment which target do you mean

Reply:
ok